src/common.rs

```diff
@@ -517,6 +519,47 @@ pub fn generate_keypair() -> color_eyre::eyre::Result<KeyPairProperties> {
     Ok(key_pair_properties)
 }
 
+fn compute_hash(bytes: &[u8]) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+    hasher.update(&bytes);
+
+    let result = hasher.finalize();
+    let mut hash = [0u8; 32];
+    hash.copy_from_slice(&result[..]);
+
+    hash
+}
+
+pub fn print_full_signed_transaction(transaction: near_primitives::transaction::SignedTransaction) {
+    eprintln!("{:<25} {}\n", "signature:", transaction.signature);
+    crate::common::print_full_unsigned_transaction(transaction.transaction);
+}
+
+pub fn print_full_unsigned_transaction(transaction: near_primitives::transaction::Transaction) {
+    let bytes = transaction
+        .try_to_vec()
+        .expect("Transaction is not expected to fail on serialization");
+    eprintln!("transaction payload (without signature) hash:");
+    eprintln!(
+        "{:<25} {}",
+        "SHA256 hash(hex):",
+        hex::encode(&compute_hash(&bytes))
```

Comment:
You don't need to implement the `compute_hash` function, Transaction type has [`get_hash_and_size()` method](https://docs.rs/near-primitives/latest/near_primitives/transaction/struct.Transaction.html#method.get_hash_and_size), just use it (and remove the method and explicit `sha2` dependency.